app/controllers/admin/books_controller.rb

```diff
@@ -5,4 +5,37 @@ def index
     @pagy, @books = pagy Book.latest_books,
                          items: Settings.max_page_user
   end
+
+  def new
+    @book = Book.new
+  end
+
+  def create
+    @book = Book.new book_params
+    if @book.save
+      @book.categories_books.insert_all(build_params)
+      flash[:message] = t "book.add_ss"
+      redirect_to admin_books_path
+    else
+      flash[:danger] = t "book.error"
+      render :new
+    end
+  end
+
+  def book_params
+    params.require(:book).permit :name, :image, :description,
+                                 :author_id, :publisher_id, :quantity
+  end
+
+  def categories_books_params
+    params.require(:book).permit category_book_ids: []
+  end
+
+  def build_params
+    arr = categories_books_params
+    arr[:category_book_ids].shift
+    arr[:category_book_ids].map do |category_book_id|
```

Comment:
categories_books_params.shift.map do |category_book_id|

Reply:
chô này em sửa theo c mà không được, cái categories_books_params.shift sẽ in ra kiểu {category_book_ids => [1,2,3,4,5]} trả về false, chỗ này em dùng biến phụ để gán giá trị sau khi xoá mới loại bỏ phần tử NULL ở đầu được @ngocvt-0484